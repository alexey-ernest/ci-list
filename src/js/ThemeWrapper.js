/**
 * Wraps component with theme provider to avoid Material UI errors.
 */

import React, {Component, PropTypes} from 'react';

import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {
  lightBlue500,
  lightBlue700,
  grey400,
  pinkA200,
  grey200,
  grey500,
  darkBlack,
  white,
  grey300,
  fullBlack
} from 'material-ui/styles/colors';
import {fade} from 'material-ui/utils/colorManipulator';


// More on Colors: http://www.material-ui.com/#/customization/colors
var muiTheme = getMuiTheme({
  palette: {
    primary1Color: lightBlue500,
    primary2Color: lightBlue700,
    primary3Color: grey400,
    accent1Color: pinkA200,
    accent2Color: grey300,
    accent3Color: grey500,
    textColor: darkBlack,
    alternateTextColor: white,
    canvasColor: white,
    borderColor: grey300,
    disabledColor: fade(darkBlack, 0.3),
    pickerHeaderColor: lightBlue500,
    clockCircleColor: fade(darkBlack, 0.07),
    shadowColor: fullBlack,
  }
});

export default class ThemeWrapper extends Component {

  static childContextTypes = {
    muiTheme: React.PropTypes.object.isRequired
  };

  getChildContext() {
    return {muiTheme: muiTheme};
  }

  render() {
    return (
      <MuiThemeProvider muiTheme={muiTheme}>
        {this.props.children}
      </MuiThemeProvider>
    );
  }
}
