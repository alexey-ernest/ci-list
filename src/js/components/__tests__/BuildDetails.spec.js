
import React from 'react';
import renderer from 'react-test-renderer';

jest.mock('react-dom');

jest.mock('material-ui/CircularProgress', () => {
  return () => {
    return (
      <p>CircularProgress</p>
    );
  }
});

jest.mock('../../actions/JobActionCreators', () => {
  return {
    requestJobBuild: jest.fn()
  };
});

jest.mock('../../../sass/modules/build-details.sass', () => {
  return {};
});


import ThemeWrapper from '../../ThemeWrapper';
import BuildDetails from '../BuildDetails.react';


describe('BuildDetails component', () => {

  it('should render correctly when there is no build data', () => {
    const component = renderer.create(
      <ThemeWrapper>
        <BuildDetails job={{}} />
      </ThemeWrapper>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should request data for the job', () => {
    const actionCreators = require('../../actions/JobActionCreators');
    const spy = jest.fn();
    actionCreators.requestJobBuild = spy;

    const job = {
      id: 5
    };

    const component = renderer.create(
      <ThemeWrapper>
        <BuildDetails job={job} />
      </ThemeWrapper>
    );

    expect(spy.mock.calls.length).toBe(1);
    expect(spy.mock.calls[0][0]).toBe(job.id);
  });

  it('should render build details when build property specified', () => {
    const build = {
      type: 'DEBUG',
      version: '1.0.2',
      package: {
        uri: 'goo.gl/packagelink',
        size: 1024 * 1024 * 5
      }
    };
    const component = renderer.create(
      <ThemeWrapper>
        <BuildDetails job={{}} build={build} />
      </ThemeWrapper>
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

});
