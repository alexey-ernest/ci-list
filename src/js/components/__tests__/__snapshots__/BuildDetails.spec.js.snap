exports[`BuildDetails component should render build details when build property specified 1`] = `
<div>
  <div
    className={undefined}>
    <div
      className={undefined}>
      <span
        className="material-icons undefined"
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "color": "#e0e0e0",
            "display": "block",
            "fontSize": 48,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none"
          }
        }>
        build
      </span>
      DEBUG
    </div>
    <div
      className={undefined}>
      <span
        className="material-icons undefined"
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "color": "#e0e0e0",
            "display": "block",
            "fontSize": 48,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none"
          }
        }>
        plus_one
      </span>
      v
      1.0.2
    </div>
    <div
      className={undefined}>
      <span
        className="material-icons undefined"
        onMouseEnter={[Function anonymous]}
        onMouseLeave={[Function anonymous]}
        style={
          Object {
            "color": "#e0e0e0",
            "display": "block",
            "fontSize": 48,
            "muiPrepared": true,
            "position": "relative",
            "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
            "userSelect": "none"
          }
        }>
        cloud_download
      </span>
      <a
        href="goo.gl/packagelink"
        target="_blank">
        Download (
        5.0
        MB)
      </a>
    </div>
  </div>
</div>
`;

exports[`BuildDetails component should render correctly when there is no build data 1`] = `
<div>
  <p>
    CircularProgress
  </p>
</div>
`;
